pkg/tfbridge/detailed_diff.go

```diff
@@ -370,7 +371,10 @@ func (differ detailedDiffer) calculateSetHashIndexMap(
 						path.String(), r))
 				}
 			}()
-			return tfs.SetHash(newElem)
+			setConfig, err := tfs.SetElement(newElem)
+			contract.AssertNoErrorf(
+				err, "Failed to calculate preview for element in %s: Failed to convert set element", path.String())
+			return tfs.SetHash(setConfig)
```

Comment:
SetElement and SetHash functions should probably be combined in the shim layer for clarity since the callers always seem to call SetHash(SetElement(x)) seems like.